rgw/admin/mock.go

```diff
@@ -0,0 +1,18 @@
+package admin
+
+import (
+	"net/http"
+)
+
+// MockClient is the mock of the HTTP Client
+// It can be used to mock HTTP request/response from the rgw admin ops API
+type MockClient struct {
+	// MockDo is a type that mock the Do method from the HTTP package
+	MockDo MockDoType
+}
+
+// MockDoType is a custom type that allows setting the function that our Mock Do func will run instead
+type MockDoType func(req *http.Request) (*http.Response, error)
```

Comment:
dito

Reply:
this doesn't have to be exported in any case, even if you want to offer a MockClient